### DISCUSSION TITLE
Issue Getting GaugeComponent to Properly ReSize/ Be Responsive

### DISCUSSION THREAD

**Discussion by jwolensky-uva** [antoniolago/react-gauge-component#30] [Q&A]
Hello There! Thank you so much for contributing this to the dev community. 

Im curious how you went about implementing the resizing behavior of the gauge component in your demo? Is the React code for this available somewhere to view?

The way I am attempting it is to create a container component (MUI Box) of a specific size in which the gauge fills up. This is working however I am running into a strange issue where if I make this container component display='flex' it shrinks the gauge component.

Thank you

**Comment by SebUndefined** [comment]
You have to calculate the size of the parent container (use a package or build your own hook) and send this data to ```GaugeComponent```. I am working on this today. All our gauge are in a flex container (with flex-wrap and a minimum width). This flex container is in a Panel from ReactGridLayout https://github.com/react-grid-layout/react-grid-layout so resize is quite mandatory. 

Will try to post you some code before the end of the week, as soon as I fix a little bug (small overflow in some specifics cases). Ping me if I forget, sounds a little mindfuck :D

**Reply by keenancabrera** [reply]
Hello!

I'd also like to thank you for this contribution, and I am also running into the same issue. I am calculating the width and passing to the gauge component via the style prop:
```
          <GaugeComponent
               style={{width : `${width}px`}}
          <GaugeComponent/>
```

Unfortunately, this is not updating the width of the of the SVG component itself, so the gauge remains the same width of the parent element when it was first rendered. Is there a proper way to pass the width information to the GaugeComponent so that it can resize properly?

**Comment by arvindh123** [comment]
@jwolensky-uva  @SebUndefined 
I also facing the same issue now , when display=flex , it shrinks, 
Is there any solution for this proble,

**Reply by SebUndefined** [reply]
Sorry but we ended up by creating our own gauge component using chartjs donuts chart. Less features but more robust.

**Comment by gkyorgos** [comment]
same symptoms here - the SVG on first load is out of wack until the view is refreshed. Using bootstrap and in an inner div....it doesnt correct until refresh. Any fix?

**Comment by antoniolago** [comment]
Hello everyone, sorry for the delay, I've just pushed v1.2.61 (#58) which probably solves this, can you update and test again?
